int8 id_1
int8 id_2